standard/classes.md

```diff
@@ -5481,3 +5483,129 @@ When the body of the async function terminates, the return task is moved out of
+
+All single-dimensional and jagged arrays ([§17.1](arrays.md#171-general)) are indexable sequences; multi-dimensional arrays are not! The use of indexes and ranges with arrays is described in [§12.8.11.2](expressions.md#128112-array-access).
+
+An object of type `string` is an indexable sequence.
+
+A user-defined type can provide explicit support for indexer access ([§12.8.11.3](expressions.md#128113-indexer-access)) using `System.Index` and `System.Range`. (See §indexable-sequence-expl-support-for-index and §indexable-sequence-expl-support-for-range.) If various criteria are met, an existing user-defined type that does *not* have such explicit support, shall have provided for it by the implementation implicit support for such indexer and range access. (See §indexable-sequence-impl-support-for-index and §indexable-sequence-impl-support-for-range.) In both cases, the type is recognized as being an indexable sequence type.
+
+### §indexable-sequence-support-for-index Providing support for Index
+
+#### §indexable-sequence-expl-support-for-index Explicit Index support
+
+A type having an instance indexer taking a single argument of type `System.Index`, or a first argument of that type followed by optional arguments, may be indexed as described by [§12.8.11.3](expressions.md#128113-indexer-access).
+
+> *Example*: In [§15.9](classes.md#159-indexers), there is an example defining type `BitArray`, which stores bits in an array of `int`. Individual bits are accessed for read/write via an `int` indexer. Adding an `Index` indexer that simply interprets the `Index` argument as an `int`, is simple:
+>
+> <!-- Example: {template:"standalone-console", name:"ExplicitIndexSupport", additionalFiles:["BitArrayPartial1.cs"], expectedOutput:["ba1[0] = True", "ba1[98] = False", "ba1[Index 0] = True", "ba1[^1] = True"]} -->
+> ```csharp
+> partial class BitArray
+> {
+>     public bool this[Index idx]
+>     {
+>         get
+>         {
+>             return this[idx.GetOffset(Length)];   // use the [int] indexer
+>         }
+>         set
+>         {
+>             this[idx.GetOffset(Length)] = value;  // use the [int] indexer
+>         }
+>     }
+> }
+> ```
+>
+> *end example*
+
+#### §indexable-sequence-impl-support-for-index Implicit Index support
+
+An implementation shall behave as if it provides a non-virtual instance indexer member with a single parameter of type `System.Index` for any type that meets the following criteria:
+
+- The type is countable ([§15.7.1](classes.md#1571-general)).
+- The type has an accessible instance indexer taking an argument of type `int` as its only argument.
+- The type does not have an accessible instance indexer taking a `System.Index` as its only argument, or as its first argument with the remaining arguments being optional.
+
+The provided instance indexer shall have the same get and set members with matching accessibility as the `int` indexer.
```

Comment:
The 'range' version of this called out that the return type was the same, including 'ref'. That is true here too.